compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1724,9 +1721,36 @@ public void visit(BLangTypeInit cIExpr) {
                 }
                 break;
             case TypeTags.STREAM:
-                dlog.error(cIExpr.pos, DiagnosticCode.INVALID_STREAM_CONSTRUCTOR,
-                           cIExpr.initInvocation.name);
-                resultType = symTable.semanticError;
+                if (cIExpr.initInvocation.argExprs.size() != 1) {
+                    dlog.error(cIExpr.pos, DiagnosticCode.INVALID_STREAM_CONSTRUCTOR, cIExpr.initInvocation.name);
+                    resultType = symTable.semanticError;
+                    return;
+                }
+
+                BStreamType actualStreamType = (BStreamType) actualType;
```

Comment:
double check - what if user sets () for ErrorType?

Reply:
created an issue for this.
https://github.com/ballerina-platform/ballerina-lang/issues/21596